ElasticSearch: Token credentials provided by AWS IAM Roles for Service Accounts do not take effect

Short version; Token credentials provided by 'AWS IAM Roles for Service Accounts' (IRSA) do not take effect because data-prepper is not packaged with the AWS Java SDK STS library.  Not sure if this is by design or an oversight - but could ease adoption.


Longer version...

I'm using AWS to host ElasticSearch and AKS to manage our K8S clusters.  Data-prepper is deployed to the k8s cluster and in order to access ES I have: 
- Created IAM roles and access policies for data-prepper => ES resources
- Setup [IAM Roles for Service Accounts](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) so AKS  injects the necessary environment variables and projected token config volume for my K8S ServiceAccount+Pod
- Configured data-prepper ES sync to use aws_sigv4

`com.amazon.dataprepper.plugins.sink.elasticsearch.ConnectionConfiguration.attachSigV4()` is utilising the AWS Java SDK Core provided `DefaultAWSCredentialsProviderChain`which includes the `WebIdentityTokenCredentialsProvider`.  This provider is correctly picking up that the AWK projected AWS_ROLE_ARN and AWS_WEB_IDENTITY_TOKEN_FILE env vars are available to it, but errors at runtime, logging that `aws-java-sdk-sts` needs to be available on the classpath.

To work around the issue, I rebuilt the app and docker image and added `aws-java-sdk-sts` as a dependency on the Elasticsearch plugin.  

**Expected behavior**
When IRSA configured on the pod and data-prepper configured to use aws_sigv4, data-prepper uses the IRSA provided token to authenticate to ES.

**Observed behaviour**
In my setup, `DefaultAWSCredentialsProviderChain` falls through to `EC2ContainerCredentialsProviderWrapper` which picks up the node level credentials which I have not setup to be able to access ES; so ES error logs.

e.g. log snippet- sensitive stuff <redacted>
```
Caused by: org.elasticsearch.client.ResponseException: method [GET], host [https://<myinstance>.eu-west-1.es.amazonaws.com:443], URI [/_cluster/settings?master_timeout=30s&inc │
:{"Message":"User: arn:aws:sts:<myId>:assumed-role/eksctl-<my-ns>-nodegroup-cic-NodeInstanceRole-<code>/<id> is not authorized to perform: es:ESHttpGet"} 
```

--- a/data-prepper-plugins/elasticsearch/build.gradle
+++ b/data-prepper-plugins/elasticsearch/build.gradle
@@ -36,6 +36,7 @@ dependencies {
     implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.12.3"
     implementation 'javax.ws.rs:javax.ws.rs-api:2.1.1'
     implementation "com.amazonaws:aws-java-sdk-core:1.11.1024"
+    implementation "com.amazonaws:aws-java-sdk-sts:1.11.1024"
     implementation "com.github.awslabs:aws-request-signing-apache-interceptor:b3772780da"
     implementation "io.micrometer:micrometer-core:1.7.0"
     testImplementation("junit:junit:4.13.2") {

